compiler/src/dotty/tools/dotc/typer/TypeAssigner.scala

```diff
@@ -145,7 +145,10 @@ trait TypeAssigner {
       // this is exactly what Erasure will do.
       case _ =>
         val pre = maybeSkolemizePrefix(qualType, name)
-        val mbr = qualType.findMember(name, pre)
+        var mbr = qualType.findMember(name, pre)
+        if(!mbr.exists && ctx.isJava && pre.typeSymbol.isClass)
+          mbr = ctx.javaFindMember(name,pre)
```

Comment:
Do we actually need to always call findMember first or could we do:
```suggestion
        var mbr = 
          if ctx.isJava && pre.typeSymbol.isClass then
             ctx.javaFindMember(name, pre)
           else
             qualType.findMember(name, pre)
```